akka-http-core/src/main/scala/akka/http/impl/engine/server/HttpServerBluePrint.scala

```diff
@@ -29,6 +29,24 @@ import akka.util.ByteString
 
 /**
  * INTERNAL API
+ *
+ *
+ * HTTP pipeline setup:
+ *
+ *                     +-------------+                 +-------------+                   +-----------+
+ *     HttpRequest     |  request-   |     Request-    |             |      Request-     | request-  |     ByteString
+ * | <-----------------+ Preparation <-----------------+             <-------------------+  Parsing  <---------------
+ * |                   |             |     Output      |             |      Output       |           |
+ * |                   +-------------+                 |             |                   +-----------+
+ * |                                                   |             |
+ * | Application-                                      | controller- |
+ * | Flow                                              |    Stage    |
+ * |                                                   |             |
+ * |                                                   |             |                   +-----------+
+ * |                  HttpResponse                     |             |     Response-     | renderer- |     ByteString
+ * v -------------------------------------------------->             +-------------------> Pipeline  +-------------->
+ *                                                     |             |     Rendering-    |           |
+ *                                                     +-------------+     Context       +-----------+
```

Comment:
I know it is getting big, but not including the One2One stage, websockets and TLS is slightly confusing; maybe there should at least be some indication that something is missing.